### PR TITLE
feat: add binning data operation package

### PR DESCRIPTION
**Pull request by TKaltofen**
## Summary
- Add binning data operation package (package 8 of 9) under `row_preserving/binning/`
- Implements equal-width (`bin`) and quantile (`qbin`) binning across 5 frameworks: PyArrow, Pandas, Polars Lazy, DuckDB, SQLite
- 78 tests covering base class, all frameworks, cross-framework comparison, edge cases (EC-011 to EC-015), and integration

## Test plan
- [x] All 78 binning-specific tests pass
- [x] Full tox suite green (180 passed, 13 skipped)
- [x] ruff, mypy, bandit all pass
- [x] Cross-framework comparison tests verify all frameworks match PyArrow reference

Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: mloda/community/feature_groups/data_operations/row_preserving/binning/base.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        cls._validate_n_bins(n_bins, feature_name)
+        return str(op), n_bins
+
+    def input_features(self, options: Options, feature_name: FeatureName) -> Optional[Set[Feature]]:
+        _feature_name = feature_name.name if isinstance(feature_name, FeatureName) else feature_name
+
+        prefix_patterns = self._get_prefix_patterns()
+        operation_config, source_feature = FeatureChainParser.parse_feature_name(_feature_name, prefix_patterns)
+
+        if operation_config is not None and source_feature is not None and source_feature:
```
Comment (TKaltofen) [inline mloda/community/feature_groups/data_operations/row_preserving/binning/base.py]: **Consistency with mixin**: The mixin's `input_features` calls `self._validate_in_feature_count(in_features, _feature_name)` in the string-parsed branch, but this override skips that validation when a source feature is extracted from the name pattern (line 92 returns immediately without validating). Since binning is always single-input (`MIN_IN_FEATURES = MAX_IN_FEATURES = 1`), the omission is harmless in practice, but it breaks the pattern established by the mixin.

**Suggestion:** Either add the validation call for consistency, or add a short comment explaining why it is intentionally omitted (e.g. "single-input guaranteed by regex pattern").
